File: config/paths.py
from __future__ import annotations
import os
from pathlib import Path

P38_PROJECT_ROOT = Path(os.getenv("P38_PROJECT_ROOT", "/data3/Group8/gonglihao/项目/p38"))
P38_LEGACY_ROOT = Path(os.getenv("P38_LEGACY_ROOT", "/data3/Group8/gonglihao/项目/p38"))
P38_OUTPUT_ROOT = Path(os.getenv("P38_OUTPUT_ROOT", str(P38_LEGACY_ROOT / "p38_publication_outputs")))
P38_THREADS = int(os.getenv("NSLOTS", os.getenv("P38_THREADS", "4")))
RANDOM_SEED = int(os.getenv("P38_RANDOM_SEED", "1234"))
for variable in ("OMP_NUM_THREADS", "OPENBLAS_NUM_THREADS", "MKL_NUM_THREADS", "NUMEXPR_NUM_THREADS"):
    os.environ[variable] = str(P38_THREADS)

def p38_path(*parts: str) -> str:
    return str(P38_PROJECT_ROOT.joinpath(*parts))

def legacy_path(*parts: str) -> str:
    return str(P38_LEGACY_ROOT.joinpath(*parts))

def output_path(*parts: str) -> str:
    path = P38_OUTPUT_ROOT.joinpath(*parts)
    path.parent.mkdir(parents=True, exist_ok=True)
    return str(path)

File: config/paths.R
p38_project_root <- Sys.getenv("P38_PROJECT_ROOT", "/data3/Group8/gonglihao/项目/p38")
p38_output_root <- Sys.getenv("P38_OUTPUT_ROOT", file.path(p38_project_root, "p38_publication_outputs"))
p38_upstream_root <- Sys.getenv("P38_UPSTREAM_ROOT", "/data3/Group8/gonglihao/20250507p38-up")
p38_workers <- as.integer(Sys.getenv("NSLOTS", Sys.getenv("P38_WORKERS", "4")))
if (is.na(p38_workers) || p38_workers < 1L) p38_workers <- 1L
Sys.setenv(OMP_NUM_THREADS=p38_workers, OPENBLAS_NUM_THREADS=p38_workers, MKL_NUM_THREADS=p38_workers)
p38_path <- function(...) file.path(p38_project_root, ...)
legacy_path <- p38_path
upstream_path <- function(...) file.path(p38_upstream_root, ...)
output_path <- function(...) {
  path <- file.path(p38_output_root, ...)
  dir.create(dirname(path), recursive=TRUE, showWarnings=FALSE)
  path
}
